lib/threema/receive/delivery_receipt.rb

```diff
@@ -17,30 +17,23 @@ class DeliveryReceipt
 
       def initialize(content:, **)
         @content = content
-        return unless type_by(content).present?
-
         @status = type_by(content)
+        return unless @status
+
         @timestamp = Time.now.utc.to_i
-        @message_ids = extract_message_ids(content)
+        @message_ids = extract_message_ids(content.slice(1, content.length - 1))
```

Comment:
Well, I would have expected that to stay part of `extract_message_ids` but it is fine for me.

Reply:
it could have, but passing this in and then naming it as `message_ids_payload` in the params of the method seems about the same as assigning it to a local variable in the method itself.